client/jetpack-cloud/sections/partner-portal/issue-multiple-licenses-form/index.tsx

```diff
@@ -56,24 +59,40 @@ export default function IssueMultipleLicensesForm( {
 
 				! allProducts.includes( product.slug )
 					? allProducts.push( product.slug )
-					: allProducts.splice( selectedProducts.indexOf( product.slug ), 1 );
+					: allProducts.splice( selectedProductSlugs.indexOf( product.slug ), 1 );
 
 				return allProducts;
 			} );
 		},
-		[ dispatch, selectedProducts ]
+		[ dispatch, selectedProductSlugs ]
 	);
 
 	useEffect( () => {
 		// In the case of a bundle, we want to take the user immediately to the next step since
 		// they can't select any additional item after selecting a bundle.
-		if ( selectedProducts.find( ( product ) => isJetpackBundle( product ) ) ) {
+		if ( selectedProductSlugs.find( ( product ) => isJetpackBundle( product ) ) ) {
 			issueLicenses();
 		}
-	}, [ selectedProducts ] );
+	}, [ selectedProductSlugs, issueLicenses ] );
 
 	const selectedSiteDomain = selectedSite?.domain;
 
+	const disabledProductSlugs = selectedProductSlugs
+		// Get the product objects corresponding to the selected product slugs
+		.map( ( selectedProductSlug ) =>
+			allProducts?.find( ( product ) => product.slug === selectedProductSlug )
```

Comment:
There's some O(n^2) computation in here, but n is small and essentially fixed in size.